lib/sgtn-client/api/base.rb

```diff
@@ -0,0 +1,92 @@
+# 
+#  Copyright 2019-2022 VMware, Inc.
+#  SPDX-License-Identifier: EPL-2.0
+#
+
+require 'multi_json'
+
+module SgtnClient
+
+  class Base
+    class << self
+
+      def getTranslation(component, key, locale)
+        locale = SgtnClient::LocaleUtil.get_best_locale(locale)
+        items = get_cs(component, locale)
+        if items.nil? || items["messages"].nil?
+          nil
+        else
+          items["messages"][key]
+        end
+      end
+
+      def compareSource(component, key, default_locale, source, translation)
+        SgtnClient.logger.debug "[Base][compareSource]component=#{component},key=#{key},default_locale=#{default_locale},source=#{source},translation=#{translation}"
+        items = get_cs(component, default_locale)
+        if items.nil? || items["messages"].nil?
+          translation
```

Comment:
what do you mean 'because get_cs(component, default_locale) returns incorrect value'? as long as the return value is not empty, the it won't enter if block, right?

Reply:
You can test it yourself. Add `SgtnClient.logger.debug items` after line 25. It will return something like this:

`{"source"=>{"some_key_1"=>"some_value_1", "some_key_2"=>"some_value_2", ...}}`

so items["messages"].nil? is always true.

Reply:
By the way, this is part of "Part 1" of this PR.

Reply:
> You can test it yourself. Add `SgtnClient.logger.debug items` after line 25. It will return something like this:
> 
> `{"source"=>{"some_key_1"=>"some_value_1", "some_key_2"=>"some_value_2", ...}}`
> 
> so `items["messages"].nil?` is always true.

Where  `{"source"=>{"some_key_1"=>"some_value_1", "some_key_2"=>"some_value_2", ...}}` is from, seems it's invalid data.

Reply:
> By the way, saving the cache for default_locale is part of "Part 1" of this PR, which is independent of the source comparison logic. Please resolve this issue in this PR even if you decide to move source-comparison changes into another PR.

Saving the cache for default_locale already exist in get_cs, if the request locale is source_locale, the cache will build up.

Reply:
get_cs(component, default_locale) returns `{"source"=>{"some_key_1"=>"some_value_1", "some_key_2"=>"some_value_2", ...}}`

Please add SgtnClient.logger.debug items after line 25 and paste what you get here.